pkg/machinehandler/machinehandler.go

```diff
@@ -126,6 +140,20 @@ func (m *MachineHandler) getAPIGroupPreferredVersion(apiGroup string) (string, e
 	return "", ErrApiGroupNotFound
 }
 
+func isMachineCRDPresent(ctx context.Context, cl client.Client) (bool, error) {
+	unstructuredCRD := &unstructured.Unstructured{}
+	unstructuredCRD.SetGroupVersionKind(v1.SchemeGroupVersion.WithKind("CustomResourceDefinition"))
+
+	err := cl.Get(ctx, client.ObjectKey{Name: "machines.machine.openshift.io"}, unstructuredCRD)
```

Comment:
I think this needs to come after the API group preferred version check. Also, the rest of the checks are not specifically referring to the Machine API group, since the approver can work with different API groups